packages/common/src/store/ui/modals/reducers.ts

```diff
@@ -0,0 +1,37 @@
+import { Action, combineReducers, Reducer } from '@reduxjs/toolkit'
+
+import { createChatModalReducer } from './create-chat-modal'
+import { BaseModalState } from './createModal'
+import { inboxUnavailableModalReducer } from './inbox-unavailable-modal'
+import { leavingAudiusModalReducer } from './leaving-audius-modal'
+import parentReducer, { initialState } from './parentSlice'
+import { Modals, ModalsState } from './types'
+
+/**
+ * Create a bunch of reducers that do nothing, so that the state is maintained and not lost through the child reducers
+ */
+const noOpReducers = Object.keys(initialState).reduce((prev, curr) => {
```

Comment:
I _think_ I understand why this is needed, but it seems a little messy? Is this something that goes away if/when all the modals move to the new strategy?

Reply:
It's not my favorite either. But yeah it's basically stubbing out "child" slice reducers as no-op passthroughs, although somewhat hackily using the initial state to determine what keys to create values for. If we had child slices for each modal, this could go away, but I don't see that happening. One other method would be to explicitly make these reducers rather than using the loop - not sure if that helps it feel less messy or more though